src/common/bsod.cpp

```diff
@@ -484,7 +484,7 @@ static signed char *tsk_name = 0;
 extern "C" void vApplicationStackOverflowHook(TaskHandle_t xTask, signed char *pcTaskName) {
     tsk_hndl = xTask;
     tsk_name = pcTaskName;
-    if (pcTaskName != nullptr && strlen((const char *)pcTaskName) > 20)
+    if (pcTaskName != nullptr && strlen((const char *)pcTaskName) < 20)
```

Comment:
I'd recommend strnlen to avoid buffer overrun in case of pcTaskName pointing to something bad